tests/integration/targets/aci_bfd_interface_policy/tasks/main.yml

```diff
@@ -0,0 +1,229 @@
+      - update_bfd_interface_pol.current.0.bfdIfPol.attributes.name == "ansible_bfd_interface_policy"
+      - update_bfd_interface_pol.current.0.bfdIfPol.attributes.descr == "Ansible BFD Interface Policy"
+      - update_bfd_interface_pol.current.0.bfdIfPol.attributes.adminSt == "disabled"
+      - update_bfd_interface_pol.current.0.bfdIfPol.attributes.detectMult == "5"
+      - update_bfd_interface_pol.current.0.bfdIfPol.attributes.minRxIntvl == "500"
+      - update_bfd_interface_pol.current.0.bfdIfPol.attributes.minTxIntvl == "270"
+
+  # Added another BFD Interface policy
+  - name: Add a new  BFD Interface policy - normal mode
+    cisco.aci.aci_bfd_interface_policy:
+      <<: *add_bfd_interface_pol
+      name: ansible_bfd_interface_policy_2
+      state: present
+    register: add_bfd_interface_pol_2
+
+  - name: Verify BFD Interface Policy creation
+    assert:
+      that:
+      - add_bfd_interface_pol_2 is changed
+      - add_bfd_interface_pol_2.previous == []
+      - add_bfd_interface_pol_2.current.0.bfdIfPol.attributes.dn == "uni/tn-ansible_tenant/bfdIfPol-ansible_bfd_interface_policy_2"
+      - add_bfd_interface_pol_2.current.0.bfdIfPol.attributes.name == "ansible_bfd_interface_policy_2"
+      - add_bfd_interface_pol_2.current.0.bfdIfPol.attributes.descr == "Ansible BFD Interface Policy"
+      - add_bfd_interface_pol_2.current.0.bfdIfPol.attributes.adminSt == "enabled"
+      - add_bfd_interface_pol_2.current.0.bfdIfPol.attributes.detectMult == "3"
+      - add_bfd_interface_pol_2.current.0.bfdIfPol.attributes.minRxIntvl == "50"
+      - add_bfd_interface_pol_2.current.0.bfdIfPol.attributes.minTxIntvl == "50"
+
+  - name: Query all BFD Interface policies in a specific tenant
+    cisco.aci.aci_bfd_interface_policy:
+      <<: *aci_info
+      tenant: ansible_tenant
+      state: query
+    register: query_all_result
+
+  - name: Verify querying all BFD Interface Policies
+    assert:
+      that:
+      - query_all_result is not changed
+      - query_all_result.current.0.fvTenant.children | length == 2
+      - query_all_result.current.0.fvTenant.children[0].bfdIfPol.attributes.name == "ansible_bfd_interface_policy_2"
+      - query_all_result.current.0.fvTenant.children[1].bfdIfPol.attributes.name == "ansible_bfd_interface_policy"
+
+  - name: Query 'ansible_bfd_interface_policy' BFD Interface policies in a specific tenant
+    cisco.aci.aci_bfd_interface_policy:
+      <<: *aci_info
+      tenant: ansible_tenant
+      name: ansible_bfd_interface_policy
+      state: query
+    register: query_result
+
+  - name: Verify querying'ansible_bfd_interface_policy BFD' Multihop Interface Policy
+    assert:
+      that:
+      - query_result is not changed
+      - query_result.current.0.bfdIfPol.attributes.dn == "uni/tn-ansible_tenant/bfdIfPol-ansible_bfd_interface_policy"
+      - query_result.current.0.bfdIfPol.attributes.name == "ansible_bfd_interface_policy"
+
+  - name: Remove a BFD Interface policy
```

Comment:
please add checkmode and idempotency for removal?

Reply:
done